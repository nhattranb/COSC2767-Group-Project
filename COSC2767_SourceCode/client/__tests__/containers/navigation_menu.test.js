/*
 *   RMIT University Vietnam
 *   Course: COSC2767 Systems Deployment and Operations
 *   Semester: 2024C
 *   Assessment: Assignment 2
 *   Author: Le Dinh Ngoc Quynh
 *   ID: s3791159
 *   Created  date: 25 Dec 2024
 *   Last modified: 25 Dec 2024
 */

import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import NavigationMenu from '../../app/containers/NavigationMenu';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('NavigationMenu component', () => {
  let store;

  const initialState = {
    menu: {
      isMenuOpen: false,
      isBrandOpen: false
    },
    brand: {
      storeBrands: []
    },
    category: {
      storeCategories: []
    },
    authentication: {
      authenticated: false
    },
    account: {
      user: null
    }
  };

  beforeEach(() => {
    store = mockStore(initialState);
    store.dispatch = jest.fn();
  });

  it('renders correctly for unauthenticated user', () => {
    const unauthenticatedStateWithNavigation = {
      ...initialState,
      navigation: {
        isMenuOpen: false
      }
    };
    store = mockStore(unauthenticatedStateWithNavigation);

    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavigationMenu />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly for authenticated user', () => {
    const authenticatedState = {
      ...initialState,
      authentication: {
        authenticated: true
      },
      account: {
        user: {
          role: 'ROLE_MEMBER'
        }
      }
    };
    store = mockStore(authenticatedState);

    const authenticatedStateWithNavigation = {
      ...authenticatedState,
      navigation: {
        isMenuOpen: false
      }
    };
    store = mockStore(authenticatedStateWithNavigation);

    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavigationMenu />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with open menu', () => {
    const openMenuState = {
      ...initialState,
      menu: {
        isMenuOpen: true,
        isBrandOpen: false
      }
    };
    store = mockStore(openMenuState);

    const openMenuStateWithNavigation = {
      ...openMenuState,
      navigation: {
        isMenuOpen: true
      }
    };
    store = mockStore(openMenuStateWithNavigation);
    
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavigationMenu />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
}); 