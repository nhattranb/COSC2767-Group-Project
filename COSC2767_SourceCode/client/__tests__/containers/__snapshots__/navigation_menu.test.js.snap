// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu component renders correctly for authenticated user 1`] = `
<DocumentFragment>
  <div
    class="navigation-menu"
  >
    <div
      class="menu-header"
    />
    <div
      class="menu-body"
    >
      <div
        class="container"
      >
        <h3
          class="menu-title text-uppercase"
        >
          Shop By Category
        </h3>
        <nav
          role="navigation"
        >
          <ul
            class="menu-list"
          />
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NavigationMenu component renders correctly for unauthenticated user 1`] = `
<DocumentFragment>
  <div
    class="navigation-menu"
  >
    <div
      class="menu-header"
    />
    <div
      class="menu-body"
    >
      <div
        class="container"
      >
        <h3
          class="menu-title text-uppercase"
        >
          Shop By Category
        </h3>
        <nav
          role="navigation"
        >
          <ul
            class="menu-list"
          />
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NavigationMenu component renders correctly with open menu 1`] = `
<DocumentFragment>
  <div
    class="navigation-menu"
  >
    <div
      class="menu-header"
    >
      <button
        aria-label="close the menu"
        class="input-btn md icon-only icon-left border-0"
        style="border-radius: 3px;"
        type="button"
      >
        <div
          class="btn-icon"
        >
          <span
            aria-hidden="true"
            class="close-icon"
          />
        </div>
      </button>
    </div>
    <div
      class="menu-body"
    >
      <div
        class="container"
      >
        <h3
          class="menu-title text-uppercase"
        >
          Shop By Category
        </h3>
        <nav
          role="navigation"
        >
          <ul
            class="menu-list"
          />
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
